Okta/OktaJWTValidator.swift

```diff
@@ -0,0 +1,246 @@
+/*
+ * Copyright (c) 2017, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import Foundation
+import Heimdall
+
+open class OktaJWTValidator: NSObject {
+    let jwt: OktaJWT
+    
+    init(_ token: OktaJWT) { self.jwt = token }
+    
+    func validateSignature(callback: @escaping (Bool?, OktaError?) -> Void) {
+        // Okta base64 URL encodes the modulus and exponent values as part of the key
+        // We need to decode them.
+        
+        getKeyForToken { response in
+            if let key = response {
+                // Get modulus and exponent from key
+                let decodedModulus = Utils.base64URLDecode(key["n"] as? String)
+                let decodedExponent = Utils.base64URLDecode(key["e"] as? String)
+                
+                if decodedModulus == nil || decodedExponent == nil {
+                    callback(nil, .jwtValidationError(error: "Modulus or exponent from key could not be validated"))
+                    return
+                }
+                
+                // Create Heimdall component
+                // This creates a public/private keypair based on the modulus and exponent.
+                let signatureValidator = Heimdall(publicTag: "okta-appauth-sdk-keys", publicKeyModulus: decodedModulus!, publicKeyExponent: decodedExponent!)
```

Comment:
Set `publicTag` to be constant. Investigate making it configurable by the `Okta.plist` file